modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -98,7 +102,10 @@ object ConstructionService {
 
     def parseUnsignedTransaction(hex: Hex): EitherT[F, ConstructionError, ConstructionParse.ParseResult] =
       for {
-        unsignedTransaction <- KryoSerializer[F].deserialize[Transaction](hex.toBytes).toEitherT.leftMap(_ => MalformedTransaction)
+        unsignedTransaction <- JsonSerializer
+          .deserialize[DAGTransaction](hex.toBytes)
+          .toEitherT
+          .leftMap(_ => MalformedTransaction)
         operations <- transactionToOperations(unsignedTransaction).toEitherT
       } yield ConstructionParse.ParseResult(operations, none)
 
```

Comment:
you need to cover now the newly added function (see below)

Reply:
See comments below.